### PR TITLE
UTOPIA-530: Postgres Backup Containers

### PR DESCRIPTION
**Pull request by akroon3r**
# Description

This PR includes the following proposed change(s):

UTOPIA-530: Postgres Backup Containers

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: openshift/backup/backup-build.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        type: Docker
+        dockerStrategy:
+          from:
+            kind: DockerImage
+            name: ${BASE_IMAGE_FOR_BUILD}
+          dockerfilePath: ${DOCKER_FILE_PATH}
+      output:
+        to:
+          kind: ImageStreamTag
+          name: ${NAME}:${OUTPUT_IMAGE_TAG}
```
Comment (kushal-arora-fw) [inline openshift/backup/backup-build.yaml:38]: do we need to create an ImageStreamTag as an object explicitly (like ImageStream), or maybe it gets created if the stream is created?

File: openshift/backup/backup-build.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  - name: GIT_REPO_URL
+    displayName: Git Repo URL
+    description: The URL to your GIT repo.
+    required: true
+    value: https://github.com/BCDevOps/backup-container.git
+  - name: GIT_REF
+    displayName: Git Reference
+    description: The git reference or branch.
+    required: true
+    value: master
```
Comment (kushal-arora-fw) [inline openshift/backup/backup-build.yaml:54]:
```suggestion
    value: main
```

File: openshift/backup/backup-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+  - name: WEEKLY_BACKUPS
+    displayName: Number of Weekly Backups to Retain
+    description: The number of weekly backup files to be retained.  Used for the `rolling` backup strategy.
+    required: false
+    value: ""
+  - name: MONTHLY_BACKUPS
+    displayName: Number of Monthly Backups to Retain
+    description: The number of monthly backup files to be retained.  Used for the `rolling` backup strategy.
+    required: false
+    value: ""
```
Comment (kushal-arora-fw) [inline openshift/backup/backup-deploy.yaml:372]: not specifying any values for these?

File: openshift/backup/backup-deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+        name: ${NAME}
+        app: ${APP_NAME}
+        role: ${ROLE}
+        env: ${TAG_NAME}
+    type: Opaque
+    stringData:
+      ftp-url: ${FTP_URL}
+      ftp-user: ${FTP_USER}
+      ftp-password: ${FTP_PASSWORD}
+      ftp-url-host: ${FTP_URL_HOST}
```
Comment (kushal-arora-fw) [inline openshift/backup/backup-deploy.yaml:96]: when I worked on backup containers, I don't recall ftp usage. Are we transferring the backup to a different location, and not the PV ?

File: openshift/backup/backup-build.yaml (hunk 1 above)
Comment (akroon3r) [reply openshift/backup/backup-build.yaml:38]: It's part of the documentation provided by https://github.com/BCDevOps/backup-container

I'm just taking the scripts from the repository that I've used to create the container (hopefully) and document the process in our repo.

Generally when you create an image stream and push an image to it, the imagestreamtag is automatically generated. If I trace my steps backwards in the tools namespace, there is a buildconfig that generates an image that is output into this imagestream, therefore the tag is created on push.

File: openshift/backup/backup-build.yaml (hunk 2 above)
Comment (akroon3r) [reply openshift/backup/backup-build.yaml]: good catch. thank you.

File: openshift/backup/backup-deploy.yaml (hunk 4 above)
Comment (akroon3r) [reply openshift/backup/backup-deploy.yaml:96]: FTP usage is a flag that is only used if a URL is present. I think if you read the first part of the readme here https://github.com/BCDevOps/backup-container you'll see that's the case?

**Review by kushal-arora-fw — APPROVED** [review]
The cron and the doc looks good.  Hope it works 🤞🤞🤞 There is a manual way to run the backup script - we can leverage that too.

We are missing some values but as mentioned - we will modify them as we understand the behaviour better.

**Pull request merged**